app/lib/link_checker/uri_checker/http_checker.rb

```diff
@@ -230,12 +236,24 @@ def make_request(method, check_ssl: true)
       if REDIRECT_STATUS_CODES.include?(response.status) && response.headers.include?("location") && !report.has_errors?
         target_uri = uri + response.headers["location"]
         subreport = ValidUriChecker
-          .new(target_uri.to_s, redirect_history: redirect_history + [uri], http_client: http_client)
-          .call
+                      .new(target_uri.to_s, redirect_history: redirect_history + [uri], http_client: http_client)
+                      .call
         report.merge(subreport)
       end
 
       response
+    rescue ArgumentError => e
+      case e.message
+      when /header (.*) CR\/LF/
+        add_problem(
+          CRLFArgumentError.new(
+            from_redirect: from_redirect?,
+          ),
+        )
+        nil
+      else
+        raise e
```

Comment:
Is there a cleaner way of doing this, by just rescuing from Argument Errors matching the specific string, and not having to specify the case where the message string doesn't match and having to raise again?

Reply:
We can't use Ruby's pattern matching here so this is the most common approach. Though it'd probably be a little cleaner with

```ruby
raise e unless e.message ~= /header (.*) CR\/LF/

add_problem(...
```

Though Googling it shows you can apply some clever approaches: https://www.honeybadger.io/blog/level-up-ruby-rescue-with-dynamic-exception-matchers/ but they might be a bit cryptic.